### PR TITLE
add softlayer getting started

### PR DESCRIPTION
**Pull request by andreaturli**

### CODE REVIEW

**Comment by jclouds-commentator** [comment]
Go to http://ee8e0296737d64dbf61d-945673528002d63806ace590c004f864.r36.cf5.rackcdn.com/ to review your changes.

**Comment by andreaturli** [comment]
hi @jdaggett do yo think it is time to merge that?

File: guides/softlayer.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+// Be sure to close the context when done
+computeServiceContext.close();
+{% endhighlight %}
+
+## SoftLayer Object Storage
+SoftLayer Object Storage is an OpenStack® based blobstore storage system.
+
+{% highlight java %}
+// Get a context with softlayer that offers the portable BlobStore API
+BlobStoreContext context = ContextBuilder.newBuilder("swift")
```
Comment (everett-toews) [inline guides/softlayer.md]: Is there a version of Softlayer Object Storage that would work with `"openstack-swift"`?

`"swift"` is deprecated and scheduled to be remove in jclouds 2.0.

Comment (andreaturli) [inline guides/softlayer.md]: Not sure about it, @everett-toews

Comment (everett-toews) [inline guides/softlayer.md]: From looking at [Auth Endpoints](http://sldn.softlayer.com/article/Introduction-Object-Storage#Authentication_Endpoints) it seems clear it's TempAuth only. 

These instructions really shouldn't be written based on deprecated code. We really don't need another thing blocking us from removing `"swift"`. 

The proper thing to do here is to update the instructions to use the [SwiftAuthenticationModule](http://jclouds-javadocs.elasticbeanstalk.com/org/jclouds/openstack/swift/v1/config/SwiftAuthenticationModule.html)

    When KeystoneProperties.CREDENTIAL_TYPE is set to tempAuthCredentials, do not use Keystone. Instead, bridge TempAuth to Keystone by faking a service catalog out of the storage url. The endpoint must be set to the TempAuth url, usually ending in auth/v1.0/.

Unfortunately that will only be in the next release.

I propose we remove Object Storage from this particular PR and update this guide with the SwiftAuthenticationModule when the next version of jclouds is released.

The alternative is to leave it as is (basically recommending users to use deprecated code) and update this guide with the SwiftAuthenticationModule when the next version of jclouds is released.

Your call @andreaturli

Comment (andreaturli) [inline guides/softlayer.md]: Thanks @everett-toews, sorry for my belated answer, but I'm not getting notifications from this thread, not sure why.

I'm not totally familiar with `SwiftAuthenticatioMode` but I'm happy to update this guide as soon as the next jclouds release will be out!

**Comment by everett-toews** [comment]
Thanks for the pull request but it's release week in jclouds and that means it's time to clean up the PR queue. This PR will be over 6 months old as of April 1. If you intend to continue work on it, please make a comment by April 2. Otherwise it will be closed on April 3.

**Comment by everett-toews** [comment]
@andreaturli The next release is out. ;) I'll leave this one up to you.

**Comment by jclouds-commentator** [comment]
Go to http://d89bcdbf5c7be57295bf-c6b8c615fe3564655ff367f2d0372343.r27.cf5.rackcdn.com/ to review your changes.

**Comment by andreaturli** [comment]
I think we can merge it now, and address the comments on swift when it will be removed. Thoughts?

File: guides/softlayer.md (hunk 1 above)
Comment (andrewgaul) [inline guides/softlayer.md]: I am fairly sure that SoftLayer should work with the modern openstack-swift provider -- can we check this before merging?

File: guides/softlayer.md [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+                 .credentials("tenantName:accessKey", "apiKey)
+                 .endpoint("https://<locationId>.objectstorage.softlayer.net/auth/v1.0")
+                 .buildView(BlobStoreContext.class);
+
+// Create a container in the default location
+context.getBlobStore().createContainerInLocation(null, container);
+
+// Use the map interface for easy access to put/get things, keySet, etc.
+context.createInputStreamMap(container).put("blob.txt", inputStream);
```
Comment (andrewgaul) [inline guides/softlayer.md]: We removed the `InputStreamMap` support long ago.

**Pull request merged**